.github/workflows/docs.yml

```diff
@@ -26,13 +36,77 @@ jobs:
       - name: Install bundler
         run: gem install bundler:1.17.2
 
-      - name: Build doc site
-        run: make docs
+      # If this was a merge to main, set the version to "head"
+      - name: Set version env var to 'head'
+        if: github.event_name == 'push'
+        run: echo "DOCS_VERSION=head" >> $GITHUB_ENV
+
+      # If this was a release, set the version number
+      - name: Set version env var
+        if: github.event_name == 'release'
+        run: echo "DOCS_VERSION=${GITHUB_REF##*/}" >> $GITHUB_ENV
+
+      - name: Set version in _config.yml
+        run: |
+          echo "version: ${{ env.DOCS_VERSION }}" >> ./docs/_config.yml
+
+      # Only need to build the full list of versions every time there is a new release
+      - name: Update version.json list
+        if: github.event_name == 'release'
+        run: ls ./gh-pages | grep -E '^v?(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$' | sed 's/^/"/;s/$/"/' | tr '\n' ',' | sed 's/.$//' | sed 's/^/[/;s/$/]/' | jq '. += ["${{ env.DOCS_VERSION }}"] | sort_by(.) | reverse | . += ["latest", "head"]' > ./docs/assets/js/versions.json
```

Comment:
I've fully validated every character of this regex 😇

Reply:
... I wonder if it's worth a comment describing what the regex is doing, and how it was derived

Reply:
Sure. It actually comes from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string I'll add a comment in the action explaining that.

Reply:
Thanks for adding the comment/variable